### PR TITLE
Add testing for validation and cleaning functions

### PR DESCRIPTION
**Pull request by JoshuaMarden**
The transform script needed testing, this added tests.

### CODE REVIEW

File: transform/test_transform.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    assert cleaned_df['scientific_name'].iloc[0] == "Quercus robur"
+    assert cleaned_df['scientific_name'].iloc[1] == "Fagus sylvatica"
+
+def test_is_valid_email(valid_emails, invalid_emails):
+    """Test the is_valid_email method"""
+    for email in valid_emails:
+        assert Validator.is_valid_email(email) == True, f"Expected True for valid email: {email}"
+
+    for email in invalid_emails:
+        assert Validator.is_valid_email(email) == False, f"Expected False for invalid email: {email}"
```
Comment (jeremiah1601) [inline transform/test_transform.py]: Could you add some tests to ensure passing in wrong data types is something that causes an error? I feel like it's always good to show that we are testing for these things

File: transform/transform.py [hunk 2]
```diff
@@ -87,9 +87,8 @@ def is_valid_email(email: str) -> bool:
         if it cannot be a valid email.
         """
         logger.info("Validating emails..")
-        pattern = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
-
-        if re.match(pattern, email):
+        pattern = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+[a-zA-Z0-9-]+\.[a-zA-Z]{2,}$'
+        if re.match(pattern, email) and not re.search(r'\.\.', email):
```
Comment (jeremiah1601) [inline transform/transform.py:102]: nice regex :)

**Review by jeremiah1601 — CHANGES_REQUESTED** [review]

File: transform/test_transform.py (hunk 1 above)
Comment (JoshuaMarden) [reply transform/test_transform.py]: will do.

Comment (JoshuaMarden) [reply transform/test_transform.py]: Made the change.

**Review by jeremiah1601 — APPROVED** [review]
Nice, always good to enforce datatypes since python doesnt do that for us

**Review by HighestAuto — COMMENTED** [review]
I don't think you are supposed to push your .venv file into main

**Pull request merged**